### PR TITLE
add usage field to response

### PR DESCRIPTION
**Pull request by johnnydevriese**
It would be nice to have usage in the response from OAI. 

https://platform.openai.com/docs/guides/structured-outputs/refusals

### CODE REVIEW

File: crates/kind-openai/src/endpoints/chat.rs [hunk 1]
```diff
@@ -199,6 +199,15 @@ impl<T> ChatCompletionChoice<T> {
 struct ChatCompletionResponseMessage<T> {
     content: T,
     refusal: Option<String>,
+    usage: Usage,
+}
+
+#[allow(dead_code)]
```
Comment (lazkindness) [inline crates/kind-openai/src/endpoints/chat.rs:205]: Get rid of this, we're just going to make it public

File: crates/kind-openai/src/endpoints/chat.rs [hunk 2]
```diff
@@ -199,6 +199,15 @@ impl<T> ChatCompletionChoice<T> {
...
+#[derive(Deserialize)]
+struct Usage {
```
Comment (lazkindness) [inline crates/kind-openai/src/endpoints/chat.rs:207]: Make this pub with accessors for each field, then re-export it from the lib.rs

File: crates/kind-openai/src/endpoints/chat.rs [hunk 3]
```diff
@@ -199,6 +199,15 @@ impl<T> ChatCompletionChoice<T> {
 struct ChatCompletionResponseMessage<T> {
     content: T,
     refusal: Option<String>,
+    usage: Usage,
```
Comment (lazkindness) [inline crates/kind-openai/src/endpoints/chat.rs:202]: Add a `usage` method on `ChatCompletionChoice` that gives us `&Usage`

**Review by lazkindness — CHANGES_REQUESTED** [review]

File: crates/kind-openai/src/endpoints/chat.rs (hunk 3 above)
Comment (johnnydevriese) [reply crates/kind-openai/src/endpoints/chat.rs]: I didn't realize that `usage` is under a different key so this wasn't the right place for it.

File: crates/kind-openai/src/endpoints/chat.rs (hunk 1 above)
Comment (johnnydevriese) [reply crates/kind-openai/src/endpoints/chat.rs]: removed.

File: crates/kind-openai/src/endpoints/chat.rs (hunk 2 above)
Comment (johnnydevriese) [reply crates/kind-openai/src/endpoints/chat.rs]: I made this pub and moved this to `lib.rs` and imported into `chat.rs`.

**Pull request merged**